src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,170 @@
+            {
+                editor.ReplaceNode(node, generator.Argument(identifier)
+                    .WithTrailingTrivia(arrayArgument.Syntax.GetTrailingTrivia())); // add any extra trivia that was after the original argument
+            }
+
+            // Return changed document
+            return editor.GetChangedDocument();
+        }
+
+        private static IArrayCreationOperation GetArrayCreationOperation(SyntaxNode node, SemanticModel model, CancellationToken cancellationToken,
+                out bool isInvoked)
+        {
+            // The analyzer only passes a diagnostic for two scenarios, each having an IArrayCreationOperation:
+            //      1. The node is an IArgumentOperation that is a direct parent of an IArrayCreationOperation
+            //      2. The node is an IArrayCreationOperation already, as it was pulled from an
+            //         invocation, like with LINQ extension methods
+
+            // If this is a LINQ invocation, the node is already an IArrayCreationOperation
+            if (model.GetOperation(node, cancellationToken) is IArrayCreationOperation arrayCreation)
+            {
+                isInvoked = true;
+                return arrayCreation;
+            }
+            // Otherwise, we'll get the IArrayCreationOperation from the argument node's child
+            isInvoked = false;
+            return (IArrayCreationOperation)model.GetOperation(node.ChildNodes().First(), cancellationToken);
+        }
+
+        private static string GetExtractedMemberName(IEnumerable<string> memberNames, string parameterName)
+        {
+            bool hasCollectionEnding = s_collectionMemberEndings.Any(x => parameterName.EndsWith(x, true, CultureInfo.InvariantCulture));
+
+            if (parameterName == "source" // for LINQ, "sourceArray" is clearer than "source"
+                || (memberNames.Contains(parameterName) && !hasCollectionEnding))
+            {
+                parameterName += "Array";
+            }
+
+            if (memberNames.Contains(parameterName))
+            {
+                int suffix = 0;
+                while (memberNames.Contains(parameterName + suffix))
+                {
+                    suffix++;
+                }
+                return parameterName + suffix;
+            }
+
+            return parameterName;
+        }
+
+        private static string GetMemberNameFromType(IArrayCreationOperation arrayCreationOperation)
+        {
+#pragma warning disable CA1308 // Normalize strings to uppercase
+            return ((IArrayTypeSymbol)arrayCreationOperation.Type).ElementType.OriginalDefinition.Name.ToLowerInvariant() + "Array";
+#pragma warning restore CA1308 // Normalize strings to uppercase
+        }
+
+        private static Accessibility GetAccessibility(ISymbol? methodSymbol)
+        {
+            if (methodSymbol is not null && Enum.TryParse(methodSymbol.GetResultantVisibility().ToString(), out Accessibility accessibility))
```

Comment:
Isn't it better/safer to switch on the result of `GetResultantVisibility` rather than relying on string equality?